src/jsutils/isAsyncIterable.js

```diff
@@ -0,0 +1,19 @@
+// @flow strict
+
+import { SYMBOL_ASYNC_ITERATOR } from '../polyfills/symbols';
+
+/**
+ * Returns true if the provided object implements the AsyncIterator protocol via
+ * either implementing a `Symbol.asyncIterator` or `"@@asyncIterator"` method.
+ */
+declare function isAsyncIterable(value: mixed): boolean %checks(value instanceof
+  AsyncIterable);
+
+// eslint-disable-next-line no-redeclare
+export default function isAsyncIterable(maybeAsyncIterable) {
+  if (maybeAsyncIterable == null || typeof maybeAsyncIterable !== 'object') {
+    return false;
+  }
+
+  return typeof maybeAsyncIterable[SYMBOL_ASYNC_ITERATOR] === 'function';
+}
```

Comment:
Can you please split this change into separate PR to make this PR smaller?

Reply:
@IvanGoncharov I moved support for returning AsyncIterables from resolvers to a new PR: https://github.com/graphql/graphql-js/pull/2712

I don't think there are any breaking changes there, and it will cut down a significant part of this PR. This includes the change of moving `isAsyncIterable` to a new file. Let me know if it should be broken down further.